src/SdfEntityCreator.cc

```diff
@@ -588,6 +588,14 @@ Entity SdfEntityCreator::CreateEntities(const sdf::Link *_link)
         linkEntity, components::WindMode(_link->EnableWind()));
   }
 
+  if (!_link->EnableGravity())
+  {
+    // If disable gravity, create a gravity component to the entity
+    // This gravity will have value 0,0,0
+    this->dataPtr->ecm->CreateComponent(
+        linkEntity, components::Gravity());
```

Comment:
Not sure why we're using a component with a Vector3 as a flag to enable/disable gravity on a link. Shouldn't we use a component that has a boolean?

Reply:
Because when I implemented this, I wasn't sure if adding another component would break the API/ABI or introducing too much overhead so I reused a seldom-used component. (Gravity component is only used for world gravity init other than here. It's also the only other gravity-related component I can reuse).

Reply:
> I think it gives the impression that you can set the gravity value on each link separately from the gravity in the world.

You know what when I first saw this component that was my first thought. If it doesn't break API/ABI then that's definitely better.

Reply:
Okay, great. Mind creating a PR. We'll hold of #2447 until then.

Reply:
If I have time I will make one but likely not very soon.

Also is there a reference/starting point for adding a component? I just remembered another reason for reusing was I felt like adding new components involved a lot of extra work.

Reply:
I drafted an example component in https://github.com/gazebosim/gz-sim/pull/2451. If this looks ok to @azeey, is @AzulRadio willing to finish that change?

Reply:
@scpeters No promise but I will see what I can do this weekend.

Reply:
thanks!